src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -135,12 +135,13 @@ public override void VisitResourceDeclarationSyntax(ResourceDeclarationSyntax sy
                     diagnostics.Write(DiagnosticBuilder.ForPosition(syntax.Type).ResourceTypesUnavailable(resourceType.TypeReference));
                 }
 
-                if (syntax.IfCondition is IfConditionSyntax ifConditionSyntax)
+                if (syntax.Value is IfConditionSyntax ifConditionSyntax)

```

Comment:
>IfConditionSyntax [](start = 36, length = 17)

I'm not particularly happy with this approach. The side effect of this will be that `IfConditionSyntax` doesn't have a type assigned to it, which breaks our existing behavior with other AST nodes.

Reply:
I wonder if it makes sense to rename it to `IfExpression` and assign type to it.

Reply:
I think we can add the type regardless of the name - just need to figure out the type check rule for it. `IfExpression` could be confused with the ternary operator, but on the other hand it does act like an expression since it's on the right-hand side of the `=`, so it's probably ok.

Reply:
Yeah sorry for causing confusion - I meant maybe we should let it inherit `ExpressionSyntax` since technically you can evaluate it and get some value, either a body object or nothing (null?). As for it's type, ideally I think it should be nullable resource (`'resourceType@apiVersion'?`), and it would be great if we can also implement the `?` operator and strict null checking so users can safely reference a conditional resource without worrying about runtime errors:
```
resource? foo '...' = if (something) {
  name: 'apple'
}

var bar = foo.name // error: foo can be null
var baz = foo?.name // 'apple' | null
```

But this is definitely something to consider post 0.3 release. For now would it work if we just assign it the type of `Body`?

Reply:
Ahh got it. Yeah, makes sense for 0.3. Yeah having script null checks would be nice, but I think there are some additional challenges, but definitely worth a discussion after 0.3.

Reply:
Btw, I think I'm going to fix this in a separate PR.

Reply:
Never mind. Just pushed out an update with the fix.

Reply:
Btw, `ExpressionSyntax` is used by completions to figure out if you're inside an expression, so I didn't inherit from it since completions are working ok without it.